adb.exe is left running in the background (`npm install -g expo-cli` fails with "EPERM: operation not permitted, unlink '...\adb.exe')

## Description

I receive an error when running `npm install -g expo-cli`

## Expected Behavior

For there not to be an error

## Observed Behavior

```
npm ERR! path C:\Users\Cory\AppData\Roaming\npm\node_modules\expo-cli\node_modules\.xdl.DELETE\binaries\windows\adb\adb.exe
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall unlink
npm ERR! Error: EPERM: operation not permitted, unlink 'C:\Users\Cory\AppData\Roaming\npm\node_modules\expo-cli\node_modules\.xdl.DELETE\binaries\windows\adb\adb.exe'
npm ERR!  { [Error: EPERM: operation not permitted, unlink 'C:\Users\Cory\AppData\Roaming\npm\node_modules\expo-cli\node_modules\.xdl.DELETE\binaries\windows\adb\adb.exe']
npm ERR!   cause:
npm ERR!    { Error: EPERM: operation not permitted, unlink 'C:\Users\Cory\AppData\Roaming\npm\node_modules\expo-cli\node_modules\.xdl.DELETE\binaries\windows\adb\adb.exe'
npm ERR!      errno: -4048,
npm ERR!      code: 'EPERM',
npm ERR!      syscall: 'unlink',
npm ERR!      path:
npm ERR!       'C:\\Users\\Cory\\AppData\\Roaming\\npm\\node_modules\\expo-cli\\node_modules\\.xdl.DELETE\\binaries\\windows\\adb\\adb.exe' },
npm ERR!   stack:
npm ERR!    'Error: EPERM: operation not permitted, unlink \'C:\\Users\\Cory\\AppData\\Roaming\\npm\\node_modules\\expo-cli\\node_modules\\.xdl.DELETE\\binaries\\windows\\adb\\adb.exe\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'unlink',
npm ERR!   path:
npm ERR!    'C:\\Users\\Cory\\AppData\\Roaming\\npm\\node_modules\\expo-cli\\node_modules\\.xdl.DELETE\\binaries\\windows\\adb\\adb.exe',
npm ERR!   parent: 'expo-cli' }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It's possible that the file was already in use (by a text editor or antivirus),
npm ERR! or that you lack permissions to access it.
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Cory\AppData\Roaming\npm-cache\_logs\2018-12-30T20_05_51_538Z-debug.log
```

## Environment

cant run because expo-cli will not install

--- a/packages/xdl/src/Android.ts
+++ b/packages/xdl/src/Android.ts
@@ -309,7 +309,7 @@ export async function upgradeExpoAsync(url?: string): Promise<boolean> {
 export async function assertDeviceReadyAsync() {
   const genymotionMessage = `https://developer.android.com/studio/run/device.html#developer-device-options. If you are using Genymotion go to Settings -> ADB, select "Use custom Android SDK tools", and point it at your Android SDK directory.`;
 
-  if (!await _isDeviceAuthorizedAsync()) {
+  if (!(await _isDeviceAuthorizedAsync())) {
     throw new Error(
       `This computer is not authorized to debug the device. Please follow the instructions here to enable USB debugging:\n${genymotionMessage}`
     );
@@ -351,9 +351,9 @@ async function _openUrlAsync(url: string) {
 }
 
 async function attemptToStartEmulatorOrAssertAsync() {
-  if (!await _isDeviceAttachedAsync()) {
+  if (!(await _isDeviceAttachedAsync())) {
     // If no devices or emulators are attached we should attempt to open one.
-    if (!await maybeStartAnyEmulatorAsync()) {
+    if (!(await maybeStartAnyEmulatorAsync())) {
       const genymotionMessage = `https://developer.android.com/studio/run/device.html#developer-device-options. If you are using Genymotion go to Settings -> ADB, select "Use custom Android SDK tools", and point it at your Android SDK directory.`;
       throw new Error(
         `No Android connected device found, and no emulators could be started automatically.\nPlease connect a device or create an emulator (https://docs.expo.io/versions/latest/workflow/android-studio-emulator).\nThen follow the instructions here to enable USB debugging:\n${genymotionMessage}`
@@ -368,7 +368,7 @@ async function openUrlAsync(url: string, isDetached: boolean = false): Promise<v
     await attemptToStartEmulatorOrAssertAsync();
 
     let installedExpo = false;
-    if (!isDetached && !await _isExpoInstalledAsync()) {
+    if (!isDetached && !(await _isExpoInstalledAsync())) {
       await installExpoAsync();
       installedExpo = true;
     }
@@ -454,7 +454,7 @@ export async function startAdbReverseAsync(projectRoot: string): Promise<boolean
   ];
 
   for (let port of adbReversePorts) {
-    if (!await adbReverse(port)) {
+    if (!(await adbReverse(port))) {
       return false;
     }
   }
@@ -479,7 +479,7 @@ export async function stopAdbReverseAsync(projectRoot: string): Promise<void> {
 }
 
 async function adbReverse(port: number) {
-  if (!await _isDeviceAuthorizedAsync()) {
+  if (!(await _isDeviceAuthorizedAsync())) {
     return false;
   }
 
@@ -493,7 +493,7 @@ async function adbReverse(port: number) {
 }
 
 async function adbReverseRemove(port: number) {
-  if (!await _isDeviceAuthorizedAsync()) {
+  if (!(await _isDeviceAuthorizedAsync())) {
     return false;
   }
 
@@ -619,5 +619,14 @@ but their sizes mismatch expected ones: [dpi: provided (expected)] ${androidSpla
       )
       .join(', ')}
 See https://docs.expo.io/versions/latest/guides/splash-screens/#differences-between-environments---android for more information`);
+  }
+}
+
+export async function maybeStopAdbDaemonAsync() {
+  try {
+    await getAdbOutputAsync(['kill-server']);
+    return true;
+  } catch {
+    return false;
   }
 }

--- a/packages/xdl/src/Project.ts
+++ b/packages/xdl/src/Project.ts
@@ -2369,6 +2369,7 @@ async function _stopInternalAsync(projectRoot: string): Promise<void> {
   await stopExpoServerAsync(projectRoot);
   ProjectUtils.logInfo(projectRoot, 'expo', '\u203A Stopping Metro bundler');
   await stopReactNativeServerAsync(projectRoot);
+  await Android.maybeStopAdbDaemonAsync();
   if (!Config.offline) {
     try {
       await stopTunnelsAsync(projectRoot);

